### ISSUE TITLE
`orElseGet` argument

### ISSUE THREAD

**Issue by msuret** [JasonStorey/Optional.js#3]
Hi,

`Optional.orElseGet` currently takes another optional as argument, or at least an object with a `get` attribute. So to use it I have to write something like :
```javascript
Optional.ofNullable('foo').orElseGet({get: ()=>'bar'})
```
which looks a bit awkward.
It would probably be better if `orElseGet` took directly a function as an argument to allow a similar syntax as Java:
```javascript
Optional.ofNullable('foo').orElseGet(()=>'bar')
```

**Comment by JasonStorey** [comment]
Ah... yes, you're totally right; it should take a supplier as an argument. Thanks!

Feel free to send a PR or I can push an update tomorrow.

**Comment by JasonStorey** [comment]
Just released v1.1.1 with your suggestion. Let me know if you think of anything else.

Now, if a .get() method is not present, it checks to see if the provided argument is a function, and uses that instead.

**Comment by msuret** [comment]
That's perfect, thanks for the good work !
